tools/get_cc_files.bzl

```diff
@@ -6,41 +6,41 @@ FilesInfo = provider(
     },
 )
 
-def _get_hdrs(ctx):
-    files = []
+def get_cc_hdrs(ctx):
+    hdrs = []
 
     if hasattr(ctx.rule.attr, "hdrs"):
         for hdr in ctx.rule.attr.hdrs:
-            for file in hdr.files.to_list():
-                if not file.path.startswith(ctx.genfiles_dir.path):
-                    files.append(file)
-    return files
+            hdrs += [hdr for hdr in hdr.files.to_list() if hdr.is_source]
```

Comment:
Using the `is_source` property is the right way to check if a file is not generated.

Reply:
Nice find